ci/test_cpp_memcheck.sh

```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+# Copyright (c) 2023, NVIDIA CORPORATION.
+
+source "$(dirname "$0")/test_cpp_common.sh"
```

Comment:
Same question as before about the quotes, I don't think this does quite what we want. If, for instance, there are any spaces in the name `$0` the result here will be having them unquoted, which is usually the kind of thing you want to use quotes for.

Reply:
@ajschmidt8 
https://github.com/rapidsai/cudf/blob/d077c9b2a3764818fb690995f2c2867a2c024811/ci/test_python_cudf.sh#L5

Reply:
Looks like this is correct for command substitution. Here are a couple references: 
https://unix.stackexchange.com/questions/289574/nested-double-quotes-in-assignment-with-command-substitution
https://unix.stackexchange.com/questions/118433/quoting-within-command-substitution-in-bash

Reply:
Ah yes pesky command substitution rules. I never remember any of this. Thanks for the refs.